app/services/group_assignment_service.rb

```diff
@@ -8,10 +8,17 @@ def initialize(new_group_assignment_params, new_grouping_params)
   end
 
   def build_group_assignment
-    group_assignment          = GroupAssignment.new(@new_group_assignment_params)
-    group_assignment.grouping = Grouping.where(id: @grouping_id).first_or_create!(@new_grouping_params)
+    group_assignment = GroupAssignment.new(@new_group_assignment_params)
+
+    if grouping_info_valid?
+      group_assignment.grouping = Grouping.where(id: @grouping_id).first_or_create!(@new_grouping_params)
+    end
 
     group_assignment.build_group_assignment_invitation
     group_assignment
   end
+
+  def grouping_info_valid?
+    (@grouping_id != '') || (@new_grouping_params[:title] != '')
```

Comment:
Is there a reason we are guarding against the id being nil?

In ActiveRecord `where` on `nil` will just return an `[]`

Also if we wanted to check to see if the id is there I'd rather do

```ruby
@grouping_id.present?
```

Rather than check if it's not equal to empty string